Unlocks to gridview

karllindmark opened the pull request:
@peter-budo 

This looks a bit better than that `ExpandableListView` listing we had before this. Phew. :japanese_castle: 

Build: http://ci.peterscorner.co.uk/job/BF4-Intel-PR/65/console

Commit:
Unlocks --> GridView (instead of ExpandableListView)

Commit:
Update unlock image map

Commit:
Minor adjustments to the unlocks

Commit:
Minor touch-ups for unlock PR

A few minor touch-ups for the PR:

- Remove duplication in Unlock adapters
- Remove unneeded logging in BaseApiService
- REmove unneeded getView() in parent class

peter-budo commented:
For some reason on 3rd or 4th rotation kits failed

```
04-13 15:18:10.597  18109-18135/com.ninetwozero.bf4intel E/CursorWindow﹕ Could not allocate CursorWindow '/data/data/com.ninetwozero.bf4intel/databases/google_analytics_v2.db' of size 2097152 due to error -12.
04-13 15:18:10.597  18109-18135/com.ninetwozero.bf4intel E/GAV3﹕ Thread[GAThread,5,main]: Error on GAThread: android.database.CursorWindowAllocationException: Cursor window allocation of 2048 kb failed. # Open Cursors=30 (# cursors opened by this proc=30)
            at android.database.CursorWindow.<init>(CursorWindow.java:104)
            at android.database.AbstractWindowedCursor.clearOrCreateWindow(AbstractWindowedCursor.java:198)
            at android.database.sqlite.SQLiteCursor.fillWindow(SQLiteCursor.java:139)
            at android.database.sqlite.SQLiteCursor.getCount(SQLiteCursor.java:133)
            at android.database.AbstractCursor.moveToPosition(AbstractCursor.java:197)
            at android.database.AbstractCursor.moveToFirst(AbstractCursor.java:237)
            at com.google.analytics.tracking.android.PersistentAnalyticsStore.getNumStoredHits(PersistentAnalyticsStore.java:491)
            at com.google.analytics.tracking.android.PersistentAnalyticsStore.removeOldHitIfFull(PersistentAnalyticsStore.java:204)
            at com.google.analytics.tracking.android.PersistentAnalyticsStore.putHit(PersistentAnalyticsStore.java:179)
            at com.google.analytics.tracking.android.GAServiceProxy.sendQueue(GAServiceProxy.java:258)
            at com.google.analytics.tracking.android.GAServiceProxy.putHit(GAServiceProxy.java:106)
            at com.google.analytics.tracking.android.GAThread$1.run(GAThread.java:154)
            at com.google.analytics.tracking.android.GAThread.run(GAThread.java:384)
04-13 15:18:10.597  18109-18135/com.ninetwozero.bf4intel E/GAV3﹕ Thread[GAThread,5,main]: Google Analytics is shutting down.
04-13 15:18:10.597  18109-18109/com.ninetwozero.bf4intel E/AndroidRuntime﹕ FATAL EXCEPTION: main
    Process: com.ninetwozero.bf4intel, PID: 18109
    java.lang.OutOfMemoryError: pthread_create (stack size 16384 bytes) failed: Try again
            at java.lang.VMThread.create(Native Method)
            at java.lang.Thread.start(Thread.java:1029)
            at com.android.volley.RequestQueue.start(RequestQueue.java:141)
            at com.android.volley.toolbox.Volley.newRequestQueue(Volley.java:66)
            at com.android.volley.toolbox.Volley.newRequestQueue(Volley.java:78)
            at com.ninetwozero.bf4intel.base.ui.BaseLoadingFragment.onCreate(BaseLoadingFragment.java:35)
            at android.support.v4.app.Fragment.performCreate(Fragment.java:1477)
            at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:893)
            at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1104)
            at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1086)
            at android.support.v4.app.FragmentManagerImpl.dispatchCreate(FragmentManager.java:1879)
            at android.support.v4.app.FragmentActivity.onCreate(FragmentActivity.java:215)
            at com.ninetwozero.bf4intel.base.ui.BaseIntelActivity.onCreate(BaseIntelActivity.java:24)
            at com.ninetwozero.bf4intel.ui.activities.MainActivity.onCreate(MainActivity.java:44)
            at android.app.Activity.performCreate(Activity.java:5231)
            at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1087)
            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2159)
            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2245)
            at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:3738)
            at android.app.ActivityThread.access$900(ActivityThread.java:135)
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1202)
            at android.os.Handler.dispatchMessage(Handler.java:102)
            at android.os.Looper.loop(Looper.java:136)
            at android.app.ActivityThread.main(ActivityThread.java:5017)
            at java.lang.reflect.Method.invokeNative(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:515)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:779)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:595)
            at dalvik.system.NativeStart.main(Native Method)
```

karllindmark commented:
@peter-budo: Yup. There's a pull request that addresses those issues: #87 Fix database related crashes

It should be solving that type of issues (ie, memory problems on rotation)

peter-budo commented:
Also there is duplication of some items in kits:
 • Assualt class defibrillator is there twice
 • Engineer M15 AT mine twice
 • Recon claymore twice
 • Support M224 Mortar twice
 • AK 5C and carbine assignment looks same by point count
 • RFB and DMR looks same by point count
 • QBS-09 and shotgun looks same by point count

All three weapon should be single assignment, not sure where and why the other is coming from

peter-budo commented:
I do not like that none of the unlocks is grouped. Yes in someways it is interesting to see upcoming unlocks, but I'm used to grouped presentation as on battlelog.

Unlocks that I already own being overlaid with alpha layer makes it rather non-descriptive. Maybe green tick box (http://eaassets-a.akamaihd.net/bl-cdn/cdnprefix/2dc6cbf9883a8f75a3c58d8/public/battledash/images/icons/green-check-small.png) in top right corner could be more understandable/easier to acknowledge... 

As for https://github.com/karllindmark/com.ninetwozero.battlelog/pull/87 I'm fine with DB releated changes, however I do not like the handling of singular instance

Commit:
Add green tick next to completed unlocks

Commit:
Add sorting functionality to the unlocks

Commit:
Improve sorting functionality within unlock fragment

karllindmark commented:
@peter-budo: Like I mentioned on Whatsapp, a sorting functionality has been added to the unlocks. The main difference between the rough implementation (b757d33) and the slightly more polished one (8a39698) is that it's now giving the user a visual cue on what it's showing (by pre-selecting the item in the submenu).

peter-budo commented on BF4Intel/src/main/res/values/strings.xml:
```diff
@@ -210,4 +210,7 @@
   <string name="msg_user_said_hooah">You said hooah</string>
+  <string name="menu_title_sort">Sort</string>
+  <string name="label_sort_categorized">By category</string>
+  <string name="label_sort_progress">By progress</string>
```
Not sure if that is descriptive name. I was thinking `By upcoming` could be option but neither does make strong statement

karllindmark replied:
Completion?

peter-budo replied:
I though of that too, but unfortunately it may have double meaning as about to be completed or already completed. What we trying to tell the user is that sort will be done by something coming soon, he is near to finish and therefore he is expecting it.

Synonyms website returned this too me

```
nearing (adj. approaching)

advancing
approximating
coming
forthcoming
imminent
impending
oncoming
threatening
upcoming
```

I will ask more English speakers in the office for suggestions, but at the moment I'm inclined to take on `By upcoming` since it is already used by DICE in Battlepacks to show your next pack for rank and weapon

peter-budo commented:
Sorted unlocks group should reflect website order of groups example: 
 • present - recon, assault, engineer, support 
 • expected - assault, engineer, support, recon

This applies to all 3 tabs

Still duplication of equipment in kits

peter-budo commented on BF4Intel/src/main/res/layout/grid_item_unlocks.xml:
```diff
@@ -0,0 +1,56 @@
+    android:singleLine="true"
+    />
+
+  <TextView
```
I still believe this is not need here since this info will be available in details fragment.
Reasons:
  • too much info in single card
  • hard to read scores as there is not enough space 
`13,000/13,000 Recon Score` would be more readable as `13,000 / 13,000 Recon Score`
  • In many cases class score description runs out of screen example claymore `104,000/104,000 Recon Sc...`

peter-budo commented on BF4Intel/src/main/res/menu/fragment_unlocks.xml:
```diff
@@ -0,0 +1,30 @@
+        android:checkableBehavior="single"
+        android:id="@+id/menu_group_sort"
+        >
+        <item
```
Don't now why but now sort options do have radio that is rather ugly, implementation of selected state of different text and background colour would be better

peter-budo commented:
I did notice that while viewing unlocks if you change user account unlocks to not get invalidated/refreshed with appropriate results

Commit:
Add correct category sorting order

Commit:
Remove duplicates from within kits

karllindmark commented:
@peter-budo: I've corrected the category sorting order and now the sorter is also removing the duplicates per kit.
